### PR TITLE
ARROW-8132: [C++] Fix S3FileSystem tests on Windows

### PR DESCRIPTION
**Pull request by pitrou**
The new Minio version is less robust when deleting directories.

### CODE REVIEW

**Comment by kszucs** [comment]
@pitrou appveyor is failing

**Comment by pitrou** [comment]
Ok, I'm skipping some assertions on Windows.

**Comment by pitrou** [comment]
Bloody git checkout failure on GHA is back.

**Comment by pitrou** [comment]
AppVeyor build green both on my fork: https://ci.appveyor.com/project/pitrou/arrow/builds/31501196 and on the ASF account: https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/31501193

**Review by kszucs — APPROVED** [review]
LGTM, the failures are unrelated.

File: cpp/src/arrow/filesystem/s3fs_test.cc [hunk 1]
```diff
@@ -840,6 +840,15 @@ class TestS3FSGeneric : public S3TestMixin, public GenericFileSystemTest {
   bool allow_move_dir() const override { return false; }
   bool allow_append_to_file() const override { return false; }
   bool have_directory_mtimes() const override { return false; }
+  bool have_flaky_directory_tree_deletion() const override {
+#ifdef _WIN32
+    // Recent Minio versions on Windows may not register deletion of all
```
Comment (kszucs) [inline cpp/src/arrow/filesystem/s3fs_test.cc]: Created a follow-up JIRA to revisit this issue later https://issues.apache.org/jira/browse/ARROW-8134

**Pull request closed without merging**